### PR TITLE
chore: add unit tests for FetchEnvs

### PR DESCRIPTION
**Pull request by alewitt2**

### CODE REVIEW

**Comment by alewitt2** [comment]
@esatterwhite please review and we'll get mustacheTemplate rolled with the new version of RD-Core

File: lib/FetchEnvs.js [hunk 1]
```diff
@@ -181,10 +181,14 @@ module.exports = class FetchEnvs {
...
+
+      if (!configMapRef && !secretMapRef && !genericMapRef) {
+        throw new Error(`oneOf configMapRef, secretMapRef, genericMapRef must be defined. Got: ${JSON.stringify(element)}`);
+      }
+
       if (configMapRef) return this.#configMapRef(element);
-      if (secretMapRef) return this.#secretMapRef(element);
-      if (genericMapRef) return this.#genericMapRef(element);
-      return element;
+      else if (secretMapRef) return this.#secretMapRef(element);
```
Comment (esatterwhite) [inline lib/FetchEnvs.js:190]: The `else`s are not necessary with the `return`.  if the if condition passes, it terminates the function. if it doesn't, it moves on to the next

you could probably just omit `if (genericMapRef)` from the last line

```javascript
if (configMapRef) return this.#configMapRef(element);
if (secretMapRef) return this.#secretMapRef(element);
return this.#genericMapRef(element)
```

File: test/fetchEnvs-tests.js [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    }
+    return match;
+  });
+  return labelSelector ? { items: lookup } : lookup;
+}
+
+describe('fetchEnvs', function () {
+  afterEach(function () {
+    controllerObject.data.object.spec.envFrom.splice(0, controllerObject.data.object.spec.envFrom.length);
+    controllerObject.data.object.spec.env.splice(0, controllerObject.data.object.spec.env.length);
```
Comment (esatterwhite) [inline test/fetchEnvs-tests.js:113]: might be a more obvious to just assign a new array
```javascript
controllerObject.data.object.spec.envFrom = []
```

File: lib/FetchEnvs.js [hunk 3]
```diff
@@ -133,7 +133,7 @@ module.exports = class FetchEnvs {
     if (krm) {
       try {
         response = await this.api({
-          uri: krm.uri({ namespace, kind, name }),
+          uri: krm.uri({ namespace, name }),
```
Comment (esatterwhite) [inline lib/FetchEnvs.js:136]: it this just not needed, or was it causing a problem?

File: test/fetchEnvs-tests.js (hunk 2 above)
Comment (alewitt2) [reply test/fetchEnvs-tests.js:113]: was reusing the array to save garbage collector some work

Comment (alewitt2) [reply test/fetchEnvs-tests.js:113]: guess its not that big a deal in testing

Comment (alewitt2) [reply test/fetchEnvs-tests.js]: updated

File: lib/FetchEnvs.js (hunk 3 above)
Comment (alewitt2) [reply lib/FetchEnvs.js:136]: just not needed. `krm.uri()` doesnt use it and i didnt want to be confused later on

**Review by adamkingit — APPROVED** [review]

**Review by esatterwhite — APPROVED** [review]

**Pull request merged**